server/util/uuid/uuid.go

```diff
@@ -43,3 +58,51 @@ func StringToBytes(text string) ([]byte, error) {
 	}
 	return uuidBytes, nil
 }
+
+func GetHostID() (string, error) {
+	hostIDOnce.Do(
+		func() {
+			userConfigDir, hostIDError := os.UserConfigDir()
+			if hostIDError != nil {
+				// Home dir is not defined
+				return
+			}
+			configDirPath := path.Join(userConfigDir, configDirName)
+			hostIDError = os.MkdirAll(configDirPath, 0755)
+			if hostIDError != nil {
+				return
+			}
+			hostIDFilepath := path.Join(configDirPath, hostIDFilename)
+			// try to create the file to write a new ID, if it already exists this will fail
+			hostIDFile, hostIDError := os.OpenFile(hostIDFilepath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0644)
+			if hostIDError != nil {
+				if hostIDError != os.ErrExist {
+					// some other I/O error ocurred when creating the file, we can't write the ID down
+					return
+				}
+				// the file exists, read the file to get the host ID
+				hostIDFile, hostIDError := os.Open(hostIDFilepath)
+				if hostIDError != nil {
+					return
+				}
+				id, hostIDError := io.ReadAll(hostIDFile)
+				if hostIDError != nil {
+					return
+				}
+				hostID = string(id)
+				return
+			}
+			// we successfully opened the file, generate and record the host id
+			id, hostIDError := guuid.NewRandom()
+			if hostIDError != nil {
+				// read failed from rand.Reader; basically this should never happen
+				return
+			}
+			if _, hostIDError = io.WriteString(hostIDFile, id.String()); hostIDError != nil {
+				return
+			}
+			hostID = id.String()
```

Comment:
if you pull this logic out into another function you can do away with a lot of the references to hostIDError and write more natural code, then just assign the returned error to hostIDError in GetHostID().

Something like:

func getOrCreateHostID() (string, error) {
  userConfigDir, err := os.UserConfigDir()
  if err != nil {
    return "", err
  }
... etc ...
}


func GetHostID() (string, error) {
  var hostID string
  var err error 
  hostIDOnce.Do(func() {
    hostID, err = getOrCreateHostID()
  })
  return hostID, err
}

Reply:
Done, I think? I didn't move the `hostID` and `hostIDError` declarations out of their `var` block and into the function, since 
I believe that would mean they would return default values every time except the time the `once` runs.